samples/nrf_rpc/protocols_serialization/client/src/ot_shell.c

```diff
@@ -738,6 +739,253 @@ static int cmd_ot(const struct shell *sh, size_t argc, char *argv[])
+	shell_print(sh, "Response timeout: %u", config->mResponseTimeout);
+	shell_print(sh, "Max attempts: %u", config->mMaxTxAttempts);
+	shell_print(sh, "Recursion flag: %u", config->mRecursionFlag);
+	shell_print(sh, "NAT64 mode: %u", config->mNat64Mode);
+	shell_print(sh, "Service mode: %u", config->mServiceMode);
+	shell_print(sh, "Transport protocol: %u", config->mTransportProto);
+
+	return 0;
+}
+
+static int cmd_dns_client_config_set(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct in6_addr in6_addr;
+	otDnsQueryConfig config;
+	unsigned long port;
+	unsigned long timeout;
+	unsigned long attempts;
+	unsigned long recursion;
+	unsigned long nat_mode;
+	unsigned long service_mode;
+	unsigned long transport;
+	int err = 0;
+
+	if (net_addr_pton(AF_INET6, argv[1], &in6_addr)) {
+		shell_error(sh, "Invalid address");
+		return 0;
+	}
+
+	port = shell_strtoul(argv[2], 0, &err);
+	timeout = shell_strtoul(argv[3], 0, &err);
+	attempts = shell_strtoul(argv[4], 0, &err);
+	recursion = shell_strtoul(argv[5], 0, &err);
+	nat_mode = shell_strtoul(argv[6], 0, &err);
+	service_mode = shell_strtoul(argv[7], 0, &err);
+	transport = shell_strtoul(argv[8], 0, &err);
+
+	memcpy(config.mServerSockAddr.mAddress.mFields.m8, in6_addr.s6_addr, OT_IP6_ADDRESS_SIZE);
+
+	config.mServerSockAddr.mPort = (uint16_t)port;
+	config.mResponseTimeout = (uint32_t)timeout;
+	config.mMaxTxAttempts = (uint32_t)attempts;
+	config.mRecursionFlag = (otDnsRecursionFlag)recursion;
+	config.mNat64Mode = (otDnsNat64Mode)nat_mode;
+	config.mServiceMode = (otDnsServiceMode)service_mode;
+	config.mTransportProto = (otDnsTransportProto)transport;
+
+	otDnsClientSetDefaultConfig(NULL, &config);
+
+	return 0;
+}
+
+static void dns_resolve_cb(otError error, const otDnsAddressResponse *response, void *context)
+{
+	const struct shell *sh = (const struct shell *)context;
+	char hostname[128];
+	otError getter_err;
+	uint16_t index = 0;
+	otIp6Address address;
+	uint32_t ttl;
+	char addr_string[NET_IPV6_ADDR_LEN];
+
+	shell_print(sh, "DNS resolve callback error: %u", error);
+
+	if (error == OT_ERROR_NONE) {
+		getter_err = otDnsAddressResponseGetHostName(response, hostname, sizeof(hostname));
+
+		if (getter_err == OT_ERROR_NONE) {
+			shell_print(sh, "Hostname %s", hostname);
+		} else {
+			shell_error(sh, "Hostname error: %u", getter_err);
+		}
+
+		while (otDnsAddressResponseGetAddress(response, index++, &address, &ttl) ==
+		       OT_ERROR_NONE) {
+			if (!net_addr_ntop(AF_INET6, (struct in6_addr *)&address, addr_string,
+					   sizeof(addr_string))) {
+				shell_error(sh, "Failed to convert the IPv6 address");
+				return;
+			}
+
+			shell_print(sh, "Address #%u: %s ttl: %u", index, addr_string, ttl);
+		}
+	}
+}
+
+static int cmd_dns_client_resolve(const struct shell *sh, size_t argc, char *argv[])
+{
+	otError error;
+	otDnsQueryConfig config;
+
+	memcpy(&config, otDnsClientGetDefaultConfig(NULL), sizeof(otDnsQueryConfig));
```

Comment:
Done, I have just use a pointer as it will have no side effects - `otDnsClientGetDefaultConfig()` always fetches the current config version from the RPC server